contrib/IECoreArnold/include/IECoreArnold/ParameterAlgo.h

```diff
@@ -62,6 +62,7 @@ IECOREARNOLD_API void getParameters( AtNode *node, IECore::CompoundDataMap &valu
 /// or false depending on whether or not the Arnold type will be an
 /// array. Returns AI_TYPE_NONE if there is no suitable Arnold type.
 IECOREARNOLD_API int parameterType( IECore::TypeId dataType, bool &array );
+IECOREARNOLD_API int parameterType( IECore::DataPtr data, bool &array );
```

Comment:
Should be `const IECore::Data *data`, since we don't need to modify the data or own a reference to it.